### PR TITLE
PR followup for os.h false linter errors

### PR DESCRIPTION
**Pull request by AidanSun05**
This PR implements the changes proposed in issue #2368.

### CODE REVIEW

**Review by DanielaE — COMMENTED** [review]
I suggest to squash all three commits into one to keep history clean.

**Comment by AidanSun05** [comment]
Thanks for the suggestion @DanielaE - I have squashed the commits together.

File: include/fmt/core.h [hunk 1]
```diff
@@ -64,7 +64,8 @@
 #  define FMT_HAS_FEATURE(x) 0
 #endif
 
-#if defined(__has_include) && !defined(__INTELLISENSE__) && \
+#if defined(__has_include) &&                                               \
+    (!defined(__INTELLISENSE__) || (!FMT_MSC_VER || FMT_MSC_VER > 1900)) && \
```
Comment (vitaut) [inline include/fmt/core.h:68]: I don't think we need the `!FMT_MSC_VER` check here because it will be covered by `!defined(__INTELLISENSE__)`.

Comment (AidanSun05) [reply include/fmt/core.h]: You are right, `!FMT_MSC_VER` is a redundant check. I have removed it.

**Comment by vitaut** [comment]
Thanks

**Pull request merged**